sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -851,6 +855,19 @@ ProgramManager::build(ProgramPtr Program, const ContextImplPtr Context,
   return Program;
 }
 
+static ProgramManager::KernelArgMask
+createKernelArgMask(const std::vector<unsigned char> &Bytes) {
+  int SizeInBits = Bytes[0];
+
+  ProgramManager::KernelArgMask Result;
+  for (int I = SizeInBits - 1; I >= 0; --I) {
+    unsigned char Byte = Bytes[Bytes.size() - 1 - (I / CHAR_BIT)];
```

Comment:
CHAR_BIT is not appropriate here, as we really assume a sequence of 8-bit integers, not chars.